### PR TITLE
Avoid hanging by using non-blocking pipe

### PR DESCRIPTION
**Pull request by paulromano**
Some of have observed that WATTS will sometimes hang when writing output (#49). This PR should fix this by using a non-blocking pipe. However, this solution (and in fact our prior code) only works on Unix-based platforms. On Windows, we now revert to using normal `subprocess.run` instead of our special version. This means that, unfortunately, if you're on Windows, using the `show_stdout` and `show_stderr` arguments won't work (they rely on `tee_stdout` and `tee_stderr` which in turn implicitly rely on our implementation of `run`).

I don't have WATTS installed anywhere on a Windows environment so I'm hoping someone can check whether this actually works there (@samgdotson?).

### CODE REVIEW

**Review by samgdotson — COMMENTED** [review]
Thanks for taking the time to look into this!
`watts` appears successful on Windows with these updates. At the moment there are issues within `ABCE` itself that are preventing a complete run, but the current issue is resolved. If the non-blocking pipe solution still works on UNIX (can this be confirmed, perhaps @nstauff?), then I would approve (however, I do not have write access so further approval is still necessary).

Also of concern is that the nonblocking-pipe solution created a significant slowdown when run on WSL2. Is there a time difference between the original code and this PR?

**Comment by paulromano** [comment]
@samgdotson Thanks for confirming that it works on Windows. Does the slowdown you're seeing on WSL2 happen regardless of whether `show_stdout` is set to true or not?

**Comment by paulromano** [comment]
@samgdotson To follow up on your question at your meeting today: on Linux, I do not find any observable slowdown from using the output capturing here compared to just running an application directly.

**Comment by nstauff** [comment]
@paulromano @samgdotson - I tested the PyARC plugin on my MAC and it works fine. Thanks!

**Review by nstauff — APPROVED** [review]

**Comment by samgdotson** [comment]
@nstauff could you merge this PR? I do not have write access.

**Pull request merged**
